src/Service.Tests/SqlTests/GraphQLMutationTests/MultipleMutationIntegrationTests.cs

```diff
@@ -0,0 +1,429 @@
+                    message: "The field: book_id cannot be present for entity: Review at level: 2.",
+                    statusCode: DataApiBuilderException.SubStatusCodes.BadRequest.ToString()
+                );
+
+            // Test 3.
+            // For a relationship between Publisher (Referenced) - Book (Referencing) defined as publisher.id <- books.publisher_id,
+            // consider the request input for Book:
+            // 1. Publisher is the parent referenced entity - so it passes on a value for publisher_id to the referencing entity Book.
+            // 2. A value is assigned for the target (referenced entity) via relationship field ('publisher'), which will give back
+            // another value for publisher_id.
+            string createMultiplePublisherName = "createPublishers";
+            string multipleCreateOneBookWithTwoPublishers = @"mutation {
+                    createPublishers(items: [{
+                                        name: ""Human publisher"",
+                                        books: [
+                                            { title: ""Book #1"", publishers: { name: ""Alien publisher"" } }
+                                               ]
+                                          }]
+                                    )
+                    {
+                        items{
+                           id
+                           name
+                        }
+                    }
+                }";
+
+            actual = await ExecuteGraphQLRequestAsync(
+                query: multipleCreateOneBookWithTwoPublishers,
+                queryName: createMultiplePublisherName,
+                isAuthenticated: true,
+                variables: null,
+                clientRoleHeader: "authenticated");
+
+            // Validate that an appropriate exception is thrown because we have two possibly conflicting sources of
+            // values for the referencing field 'publisher_id'.
+            SqlTestHelper.TestForErrorInGraphQLResponse(
+                    actual.ToString(),
+                    message: "Found conflicting sources of values for the field: publisher_id for entity: Book at level: 2." +
+                    "Source 1: Parent entity: Publisher, Source 2: Relationship: publishers.",
+                    statusCode: DataApiBuilderException.SubStatusCodes.BadRequest.ToString()
+                );
+        }
+
+        /// <summary>
+        /// Test to validate that when one referencing column references multiple referenced columns in one referenced entity,
+        /// we fail the request during request validation because then we have conflicting sources of truth for values for
+        /// the referencing column. In such cases, the referencing column can assume the value of any referenced column which
+        /// leads to amibugites as to what value to assign to the referencing column.
+        /// </summary>
+        [TestMethod]
```

Comment:
```suggestion
        /// Test to validate that when one referencing column references multiple referenced columns in one referenced entity,
        /// we fail the request during request validation because then we have conflicting sources of truth for values for
        /// the referencing column. In these instances, the column that is being referenced can take on the value from any
        /// column it references. This can create uncertainty about which value should be assigned to the column being referenced.
        /// </summary>
        [TestMethod]
```